### ISSUE TITLE
Some games crash during upload

### ISSUE THREAD

**Issue by rafaelcgs10** [Kyraminol/3DSync#1]
I'm using an Old 3DS. 
While uploading some games like Smash Bros, Mario Kart and Mario 3D Land, the 3DSync app crashes. Unfortunately the crash message doesn't say much. I'm linking a video of the problem occurring.

[Video](https://gfycat.com/wellgroomedsentimentalbeaver)

Hope this information enough, but ask me for anything if you need.

EDIT:

I was able to upload the Mario 3D Land file after I deleted all other save files.
My guess is that the [upload endpoint](https://www.dropbox.com/developers/documentation/http/documentation#files-upload) hits the upload limit of 150mb when uploading a lot of files.

**Comment by rafaelcgs10** [comment]
So, I discovered what is causing the problem. It's the `fopen` function returning 0.
Usually `fopen` returns 0 when you try to open again a file that was already opened. However, I checked if the vector with the files to be open had any duplicated elements, but it was not the case.
Maybe there are some restrictions with `fopen` in the  3DS architecture.

The solution was just to `fclose` after all the work with the current file is done.

Also, I had some issues with Dropbox complaining about file conflict. So I also added some params to the curl call to avoid that.

I'll PR my changes.
